packages/sources/ipfs/src/endpoint/write.ts

```diff
@@ -57,16 +70,16 @@ export const execute: ExecuteWithConfig<Config> = async (request, _, config) =>
   const format = validator.validated.data.format
   const hashAlg = validator.validated.data.hashAlg
 
-  const client = create({ url: config.api.baseURL })
+  const client = create({ url: config.api?.baseURL })
   const options = { cidVersion }
 
   let cid: IPFSPath = ''
   switch (type) {
     case 'raw':
-      cid = await putFile(serialize(data, codec), client, options)
+      cid = await putFile(serialize(data as Record<string, unknown>, codec), client, options)
```

Comment:
This one is not necessarily always `Record<string, unknown>`. Might have to make it `string | Record<string, unknown>`